internal/runners/clean/run.go

```diff
@@ -58,3 +62,50 @@ func removeEnvPaths(cfg configurable) error {
 
 	return nil
 }
+
+var errDirNotEmpty = errs.New("Not empty")
+
+func removeEmptyDir(dir string) error {
+	empty, err := fileutils.IsEmptyDir(dir)
+	if err == nil && empty {
+		removeErr := os.RemoveAll(dir)
+		if err != nil {
+			return errs.Wrap(removeErr, "Could not remove directory")
+		}
+	} else if err != nil {
+		return errs.Wrap(err, "Could not check if directory is empty")
+	}
+
+	if !empty {
+		return errDirNotEmpty
+	}
+
+	return nil
+}
+
+func cleanInstallDir(dir string) error {
+	var asFiles = []string{
+		installation.InstallDirMarker,
+		constants.StateInstallerCmd + exeutils.Extension,
+		// The system directory is on MacOS only and contains the tray
+		// application files. It is safe for us to remove this directory
+		// without first inspecting the contents.
+		"system",
+	}
+
+	for _, file := range asFiles {
+		f := filepath.Join(dir, file)
+
+		var err error
+		if fileutils.IsDir(f) && fileutils.DirExists(f) {
+			err = os.RemoveAll(f)
+		} else if fileutils.FileExists(f) {
+			err = os.Remove(f)
+		}
+		if err != nil {
+			return errs.Wrap(err, "Could not clean install directory")
+		}
+	}
```

Comment:
I'd add the bin dir executables here as well, keep it all nice and central.

Also I think Windows and Linux might have some additional files? Can you make sure this is covered by an integration test which is ran for this PR?

Reply:
Linux has no additional files. Windows has an icon file which I added to the integration test but this function does not run on Windows.